src/mkindex_algo.hpp

```diff
@@ -691,38 +691,55 @@ auto parseAndStoreTaxTree(std::vector<bool>          & taxIdIsPresent,
 
 
 template <bool is_bi,
+          AlphabetEnum c_redAlph,
           typename TStringSet>
 auto generateIndex(TStringSet                       & seqs,
                    LambdaIndexerOptions       const & options)
 {
     using TIndex = std::conditional_t<is_bi,
-                                      seqan3::bi_fm_index<seqan3::text_layout::collection>,
-                                      seqan3::fm_index<seqan3::text_layout::collection>>;
+                                      seqan3::bi_fm_index<seqan3::innermost_value_type_t<decltype(seqs)>, seqan3::text_layout::collection>,
+                                      seqan3::fm_index<seqan3::innermost_value_type_t<decltype(seqs)>, seqan3::text_layout::collection>>;
+    using TRedAlph       = _alphabetEnumToType<c_redAlph>;
+
     TIndex index;
 
     myPrint(options, 1, "Generating Index...");
     double s = sysTime();
 
-    if constexpr (std::Same<seqan3::innermost_value_type_t<TStringSet>, seqan3::aa27>)
+    if constexpr (std::same_as<seqan3::innermost_value_type_t<TStringSet>, seqan3::aa27>)
     {
-        switch (options.indexFileOptions.redAlph)
+        if constexpr (std::same_as<TRedAlph, seqan3::aa10murphy>)
         {
-            case AlphabetEnum::MURPHY10:
-            {
-                auto redSeqs = seqs | seqan3::view::deep{seqan3::view::convert<seqan3::aa10murphy>};
-                index = TIndex{redSeqs};
-            } break;
-
-            case AlphabetEnum::AMINO_ACID: // no reduction
-            {
-                index = TIndex{seqs};
-            } break;
-
-            default:
-            {
-                throw std::runtime_error{"This reduced alphabet is not handled correctly."};
-            } break;
+            auto redSeqs = seqs | seqan3::views::deep{seqan3::views::convert<seqan3::aa10murphy>};
+            index = TIndex{redSeqs};
         }
+        else if constexpr (std::same_as<TRedAlph, seqan3::aa27>) // no reduction
+        {
+            index = TIndex{seqs};
+        }
+        else
+        {
+            throw std::runtime_error{"This reduced alphabet is not handled correctly."};
+        }
+
+        // switch (options.indexFileOptions.redAlph)
+        // {
+        //     case AlphabetEnum::MURPHY10:
+        //     {
+        //         auto redSeqs = seqs | seqan3::views::deep{seqan3::views::convert<seqan3::aa10murphy>};
+        //         index = TIndex{redSeqs};
+        //     } break;
+        //
+        //     case AlphabetEnum::AMINO_ACID: // no reduction
+        //     {
+        //         index = TIndex{seqs};
+        //     } break;
+        //
+        //     default:
+        //     {
+        //         throw std::runtime_error{"This reduced alphabet is not handled correctly."};
+        //     } break;
+        // }
```

Comment:
I think all of this including the next line can just be replaced with 
```cpp
index = TIndex{seqs};
```
Or not?

Reply:
You mean including the reduced version?

Reply:
Something like this:
```cpp
if constexpr (std::same_as<seqan3::innermost_value_type_t<TStringSet>, seqan3::TRedAlph>)
{
     index = TIndex{seqs};
}
else
{
    auto redSeqs = seqs | seqan3::views::deep{seqan3::views::convert<TRedAlph>};
    index = TIndex{redSeqs};
}
```

So there are no more alphabet specific things.

Reply:
Ahhh I see. Yes I think that should work, looks valid.

Reply:
Works and looks better now :slightly_smiling_face: